lib/ansible/modules/cloud/amazon/aws_kms_facts.py

```diff
@@ -0,0 +1,355 @@
+      returned: always
+      contains:
+        constraints:
+          description: Constraints on the encryption context that the grant allows.
+            See U(https://docs.aws.amazon.com/kms/latest/APIReference/API_GrantConstraints.html) for further details
+          type: dict
+          returned: always
+          sample:
+            encryption_context_equals:
+               "aws:lambda:_function_arn": "arn:aws:lambda:ap-southeast-2:012345678912:function:xyz"
+        creation_date:
+          description: Date of creation of the grant
+          type: str
+          returned: always
+          sample: 2017-04-18T15:12:08+10:00
+        grant_id:
+          description: The unique ID for the grant
+          type: str
+          returned: always
+          sample: abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234
+        grantee_principal:
+          description: The principal that receives the grant's permissions
+          type: str
+          returned: always
+          sample: arn:aws:sts::0123456789012:assumed-role/lambda_xyz/xyz
+        issuing_account:
+          description: The AWS account under which the grant was issued
+          type: str
+          returned: always
+          sample: arn:aws:iam::01234567890:root
+        key_id:
+          description: The key ARN to which the grant applies.
+          type: str
+          returned: always
+          sample: arn:aws:kms:ap-southeast-2:123456789012:key/abcd1234-abcd-1234-5678-ef1234567890
+        name:
+          description: The friendly name that identifies the grant
+          type: str
+          returned: always
+          sample: xyz
+        operations:
+          description: The list of operations permitted by the grant
+          type: list
+          returned: always
+          sample:
+            - Decrypt
+            - RetireGrant
+        retiring_principal:
+          description: The principal that can retire the grant
+          type: str
+          returned: always
+          sample: arn:aws:sts::0123456789012:assumed-role/lambda_xyz/xyz
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.ec2 import AWSRetry, camel_dict_to_snake_dict, HAS_BOTO3
+from ansible.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+
+import traceback
+
+try:
+    import botocore
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+# Caching lookup for aliases
+_aliases = dict()
+
+
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def get_kms_keys_with_backoff(connection):
+    paginator = connection.get_paginator('list_keys')
+    return paginator.paginate().build_full_result()
+
+
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def get_kms_aliases_with_backoff(connection):
+    paginator = connection.get_paginator('list_aliases')
+    return paginator.paginate().build_full_result()['Aliases']
```

Comment:
The other `get_kms_*` functions return the full response. Can you either return the full response here, or standardize on only returning the objects?